src/backend/libc/fs/syscalls.rs

```diff
@@ -2272,15 +2272,25 @@ pub(crate) fn lgetxattr(path: &CStr, name: &CStr, value: &mut [u8]) -> io::Resul
     }
 
     #[cfg(apple)]
-    unsafe {
-        ret_usize(c::getxattr(
-            path.as_ptr(),
-            name.as_ptr(),
-            value_ptr.cast::<c::c_void>(),
-            value.len(),
-            0,
-            c::XATTR_NOFOLLOW,
-        ))
+    {
+        // Passing an empty to slice to getxattr leads to ERANGE on macOS. Pass null instead.
+        // Fixes https://github.com/bytecodealliance/rustix/issues/957
```

Comment:
I have a preference for self-sufficient comments, where feasible. In this case, I believe the first line of the comment is entirely sufficient to explain what this code is doing, so the second line with the URL isn't needed.